mixin/smarty-v2@1/mixin.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * Auto-register "templates/" folder.
+ *
+ * @mixinName smarty-v2
+ * @mixinVersion 1.0.0
+ * @since 5.58
+ *
+ * @param CRM_Extension_MixInfo $mixInfo
+ *   On newer deployments, this will be an instance of MixInfo. On older deployments, Civix may polyfill with a work-a-like.
+ * @param \CRM_Extension_BootCache $bootCache
+ *   On newer deployments, this will be an instance of MixInfo. On older deployments, Civix may polyfill with a work-a-like.
+ */
+return function ($mixInfo, $bootCache) {
+  $dir = $mixInfo->getPath('templates');
+  if ($mixInfo->isActive() && file_exists($dir)) {
+    \Civi::dispatcher()->addListener('hook_civicrm_config', function () use ($dir) {
+      // Is it good or bad that this hasn't different static-guard-i-ness than the old 'hook_config' boilerplate?
+      \CRM_Core_Smarty::singleton()->addTemplateDir($dir);
```

Comment:
probably pretty neutral - I thought it was good at first - but then saw the safeguard in the function that calls this is gone now

Reply:
I did a little experiment -- adding this to `addTemplateDir()`:

```php
      if (in_array($path, $this->template_dir)) {
        throw new \RuntimeException("No duplicates!");
      }
```

Then I clicked around a bit. In regular usage, it never activated. But it did activate when toggling extensions. Which I suppose makes sense (since that gives a second/alternative path to registration).

Reply:
@totten since it seems we are adding 2 versions of the mixin the backwards compat should be with the static check by default (ie OK to change but if we have doubts then no change)